### PR TITLE
Sprint1 task Monitoring health of public websites

### PR DESCRIPTION
**Pull request by Saad-Mohsin-Khan**
This is my sprint1 task kindly review it and let me know if it needs any update.

### CODE REVIEW

File: saad/sprint1/infra/README.md [hunk 1]
```diff
@@ -32,11 +32,17 @@ This project contains the following features:
...
+The lambda handler present in web_metrics.py file calls a function read_url_FROMConstants to read constants defined in web_constants.py file and extract the url. On the extracted url we run the web_metrics function that using urllib3 and time python packages gives us information on the availability and latency of that url.
 
-This is a blank project for Python development with CDK.
+# Installation
+
+You can visit <a href=https://docs.aws.amazon.com/cdk/latest/guide/work-with.html#work-with-prerequisites>AWS CDK installation link</a>. After installing it you can simply fork this repository and follow the usage instructions provided below 
+
+# Usage
 
 The `cdk.json` file tells the CDK Toolkit how to execute your app.
```
Comment (shaharyar2021skipq) [inline saad/sprint1/infra/README.md:48]: Very well written README file, many important sections included.

File: saad/sprint1/infra/infra/infra_stack.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        #Declaring AWS Schedule Event For AWS Lambda so that it can run periodically for 5 minutes
+        lambda_schedule=aws_events.Schedule.rate(core.Duration.minutes(5))
+        #Defining the lambda handler on which AWS Events Schedule would run 
+        lambda_event_target=aws_events_targets.LambdaFunction(web_metric_lambda)
+        #Defining an AWS Event Rule that links the schedule with the target and creates a CloudWatch EventRule
+        lambda_run_rule=aws_events.Rule(self,'saadWebMetricSchedules',#id
+        description='Monitoring Website Health Every 5 mnutes',
+        schedule=lambda_schedule,
+        targets=[lambda_event_target])
+
```
Comment (shaharyar2021skipq) [inline saad/sprint1/infra/infra/infra_stack.py:29]: Good Commenting. Consider more descriptive that is already not self described by code line itself.

File: saad/sprint1/infra/lambda/.~c9_invoke_g7YmyU.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import urllib3
+from time import time
+import web_constants
+
+def lambda_handler(event,context):
+    message=read_url_FromFile('website_list.txt')
+    return message
+
```
Comment (shaharyar2021skipq) [inline saad/sprint1/infra/lambda/.~c9_invoke_g7YmyU.py]: Consider explaining input and output arguments through comments for this function

File: saad/sprint1/infra/lambda/.~c9_invoke_g7YmyU.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    return message
+    
+def read_url_FromFile(filename):
+    web_dict={}
+    with open(filename,'r') as url_file:
+        for url in url_file:
+           url_final= url.strip('\n')
+           web_dict[url_final]=web_metrics(url_final)
+    return web_dict
+
```
Comment (shaharyar2021skipq) [inline saad/sprint1/infra/lambda/.~c9_invoke_g7YmyU.py]: Good Logic. Just need to consider comments and explaining function arguments

**Review by shaharyar2021skipq — COMMENTED** [review]
Good Modular code. Comments were there that made it easy to understand but they can be considered to be improved

File: saad/sprint1/infra/README.md (hunk 1 above)
Comment (Saad-Mohsin-Khan) [reply saad/sprint1/infra/README.md:48]: Thanks

File: saad/sprint1/infra/infra/infra_stack.py (hunk 2 above)
Comment (Saad-Mohsin-Khan) [reply saad/sprint1/infra/infra/infra_stack.py:29]: Well I will look into how I can make it more descriptive. Thanks for the feedback

File: saad/sprint1/infra/lambda/.~c9_invoke_g7YmyU.py (hunk 3 above)
Comment (Saad-Mohsin-Khan) [reply saad/sprint1/infra/lambda/.~c9_invoke_g7YmyU.py]: By the way I don't know why this file was created It should not be here I will remove it. The only file containing the lambda handler is web_metrics.py

File: saad/sprint1/infra/lambda/.~c9_invoke_g7YmyU.py (hunk 4 above)
Comment (Saad-Mohsin-Khan) [reply saad/sprint1/infra/lambda/.~c9_invoke_g7YmyU.py]: This file should not be here I never created it on my system. I don't know how it got created I will look into it.

**Pull request closed without merging**